crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -36,13 +36,13 @@ def restore
 
       protected
 
-      def knife
+      def restore_chef
         [:nodes, :roles, :clients, :databags].each do |type|
           Dir.glob(@data.join("knife", type.to_s, "**", "*")).each do |file|
             file = Pathname.new(file)
             next unless file.extname == ".json"
             record = JSON.load(file.read)
-            if file.basename.to_s.include?("-default.json")
+            if file.basename.to_s.include?("-default.json") && type == :databags
               bc_name = file.basename.to_s.split("-").first
               Proposal.create(barclamp: bc_name, name: "default", properties: json)
               SchemaMigration.run_for_bc(bc_name)
```

Comment:
Replying to https://github.com/crowbar/crowbar-core/pull/144#discussion_r50018136 here (easier to get the discussion in the new PR):

if you look two lines below here, we do a `record.save`. That's a chef object that is being saved, and that will fail if chef is down. We try to deal with such exceptions elsewhere in the code (`Net::HTTPServerException` or `Errno::ECONNREFUSED`, should not be too hard to find examples).

We should just catch the exception and raise some nice "Restore failed" exception somehow.